hack/generator/pkg/codegen/pipeline_status_from_swagger.go

```diff
@@ -60,40 +59,35 @@ func augmentResourcesWithStatus(idFactory astmodel.IdentifierFactory, config *co
 
 			klog.V(1).Infof("Loaded Swagger data (%v resources, %v other types)", len(swaggerTypes.resources), len(swaggerTypes.otherTypes))
 
-			newTypes := make(astmodel.Types)
-
 			statusTypes, err := generateStatusTypes(swaggerTypes)
 			if err != nil {
 				return nil, err
 			}
 
-			found := 0
+			// put all types into a new set
+			newTypes := make(astmodel.Types)
+			// all non-resources from Swagger are added regardless of whether they are used
+			// if they are not used they will be pruned off by a later pipeline stage
+			// (there will be no name clashes here due to suffixing with "_Status")
+			newTypes.AddTypes(statusTypes.otherTypes)
+
+			matchedResources := 0
+			// find any resources and update them with status info
 			for typeName, typeDef := range types {
-				// for resources, try to find the matching Status type
 				if resource, ok := typeDef.Type().(*astmodel.ResourceType); ok {
-					var newStatus astmodel.Type
-					if statusDef, ok := statusTypes.resourceTypes.tryFind(typeName); ok {
-						klog.V(4).Infof("Swagger information found for %v", typeName)
-						newStatus = statusDef
-						found++
-					} else {
-						klog.V(4).Infof("Swagger information missing for %v", typeName)
-						// add a warning that the status is missing
-						// this will be reported if the type is not pruned
-						newStatus = astmodel.NewErroredType(nil, nil, []string{fmt.Sprintf("missing status information for %v", typeName)})
+					// find the status type (= Swagger resource type)
+					newStatus, located := statusTypes.findResourceType(typeName)
+					if located {
+						matchedResources++
 					}
-					newTypes.Add(astmodel.MakeTypeDefinition(typeName, resource.WithStatus(newStatus)))
+
+					newTypes.Add(typeDef.WithType(resource.WithStatus(newStatus)))
 				} else {
-					// other types are simply copied
 					newTypes.Add(typeDef)
 				}
 			}
 
-			// all non-resources are added regardless of whether they are used
-			// if they are not used they will be pruned off by a later pipeline stage
-			newTypes.AddAll(statusTypes.otherTypes)
-
-			klog.V(1).Infof("Found status information for %v resources", found)
+			klog.V(1).Infof("Found status information for %v resources", matchedResources)
```

Comment:
Is level 1 correct here?

Reply:
@theunrepentantgeek are you suggesting it should be lower or higher? I'd say given this log isn't very verbose that it's a 0 or 1.